src/Qowaiv.DomainModel/Collections/ImmutableCollection.Append.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Qowaiv.DomainModel.Collections;
+
+public partial class ImmutableCollection
+{
+    /// <remarks>Internal from .NET.</remarks>
+    private const int MaxCapacity = 0X7FFFFFC7;
+
+    [Pure]
+    private ImmutableCollection Append(IEnumerator iterator)
+    {
+        var count = Count;
+        var capacity = Capacity;
+        var buffer = Buffer;
+        var locker = Locker;
+        var clone = true;
+
+        if (count < capacity && buffer[count] is null)
+        {
+            lock (locker)
+            {
+                if (buffer[count] is null)
+                {
+                    while (count < capacity && iterator.MoveNext())
+                    {
+                        var item = iterator.Current;
+                        if (item is null) continue;
+                        buffer[count++] = item;
+                    }
+                    clone = count == capacity;
+                }
+            }
+        }
+
+        if (clone)
+        {
+            var copy = new object[capacity];
+            Array.Copy(buffer, copy, count);
```

Comment:
Copying the array using `Span` is usually faster in my experience. Alternatively `Buffer.BlockCopy` might also be faster.

Reply:
That can be tested. Although `List<T>` also use `Array.Copy`.